woocommerce/woocommerce-gutenberg-products-block
# Display Checkout block with a notice when adding a product sold individually that was already in the cart

Fixes #2670.

Some changes included in this PR:
* Notices generated before running the Checkout block logic are printed before. This ensures that when CartController [calls](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/main/src/StoreApi/Utilities/CartController.php#L244) `NoticeHandler::convert_notices_to_exceptions()` there are not old notices.
* Renamed `CheckoutError` to `CheckoutOrderError` so it matches its directory name. 2278519f97114a16bd7b940e8a0661e5e3a63ea9
* Replaced two strings to match the same strings in Core, so we reduce the effort to localize WC. 957152137baac567e994546a1deae848c5b9bccb

Tagging @mikejolley for review since you worked on this and will probably know if this approach makes sense or there is a better way I missed.

### Screenshots

![imatge](https://user-images.githubusercontent.com/3616980/87408529-e33de380-c5c3-11ea-82d4-13b89b7e88d7.png)

### How to test the changes in this Pull Request:

There are steps to reproduce in #2670, but I think they can be simplified to:

1. Edit a product so it can only be added once to the Cart.
![imatge](https://user-images.githubusercontent.com/3616980/87408703-27c97f00-c5c4-11ea-9ebb-c2dfdcc5472e.png)
2. Go to `https://one.wordpress.test/checkout/?add-to-cart=19` (replacing `one.wordpress.test` with your test site URL and `19` with the ID of the product you modified).
3. Verify the Checkout block appears correctly and the error message appears as a notice.

### Changelog

> Fixed a bug that was causing the Checkout block to break when adding a product sold individually that was already in the cart
